gameFunctions.js

```diff
@@ -0,0 +1,288 @@
+    //     json: true,
+    //     headers: {
+    //         Authorization: 'Bearer ' + process.env.SLACK_AUTH_TOKEN
+    //     }
+    // }, (error, response, body) => {
+    //     if (error) {
+    //         console.log(error);
+    //     } else {
+    //         const bgChannel = body.channels.find(x => x.name == 'nothing-to-see-here');
+    //         if (bgChannel) {
+    //             console.log('Found BG Channel');
+    //             if (bgChannel.is_archived) {
+    //                 console.log('BG Channel is archived, unarchiving');
+    //                 unarchiveChannel(bgChannel.id).then(() => {
+    //                     console.log('Successfully unarchived, setting up BG channel');
+    //                     setupBGChannel(roles, bgChannel.id);
+    //                 });
+    //             } else {
+    //                 setupBGChannel(roles, bgChannel.id);
+    //             }
+    //         } else {
+    //             // createChannel().then((newChannelId) => {
+    //             //     createGroup(roles.filter(x => x.role == 'bg'), newChannelId);
+    //             // })
+    //         }
+    //     }
+    // });
+}
+
+const getFormattedRoles = (players) => {
+    let roleDisplay = 'Player Roles: \n';
+    players.sort((a,b) => a.id - b.id);
+    players.forEach((player) => {
+        roleDisplay += `\t<@${player.id}>`;
+        if (player.role) {
+            roleDisplay += `\t(${player.role.toUpperCase()})`;
+        }
+        roleDisplay += `\n`;
+    });
+    return roleDisplay;
+}
+
+const unarchiveChannel = (channelId) => {
+    return new Promise((resolve, reject) => {
+        request.post({
+            url: 'https://slack.com/api/conversations.unarchive',
+            json: true,
+            headers: {
+                Authorization: 'Bearer ' + process.env.SLACK_AUTH_TOKEN
+            },
+            body: {
+                channel: channelId,
+            }
+        }, (error, response, rawBody) => {
+            console.log(rawBody);
+            if (error) {
+                console.log(error);
+                reject();
+            }
+            resolve();
+        });
+    });
+}
+
+const setupBGChannel = (players, channelId) => {
+    console.log('Removing players from BG Channel');
+    kickGroup(players, channelId).then(() => {
+        console.log('Adding players to bg channel');
+        createGroup(players.filter(x => x.role == 'bg'), channelId);
+    });
+}
+
+const notifyPlayers = (players) => {
+    players.forEach((player) => {
+        request.post({
+            url: 'https://slack.com/api/chat.postMessage',
+            json: true,
+            headers: {
+                Authorization: 'Bearer ' + process.env.SLACK_AUTH_TOKEN
+            },
+            body: {
+                channel: player.id,
+                text: getRoleText(player.role)
+            }
+        }, (error, response, rawBody) => {
```

Comment:
I think we should put a function in responseFunctions for sending a post response instead of a in-channel response.

Reply:
Agreed. Making a note. Next thing i'm going to do is actually refactoring

Reply:
I am also of the mindset "Make it work first, then make it right" lol